### PR TITLE
HIVE-25894: Table migration to Iceberg doesn't remove HMS partitions

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
During Iceberg table migration of partitioned tables we should drop the partition objects from the HMS.

### Why are the changes needed?
So we have a clean HMS metastore db

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
The existing unit test are checking the migration path

### CODE REVIEW

**Review by lcspinter — APPROVED** [review]

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java [hunk 1]
```diff
@@ -3939,8 +3939,8 @@ protected boolean getPartitionsByExprInternal(String catName, String dbName, Str
       boolean allowSql, boolean allowJdo) throws TException {
     assert result != null;
 
-    final ExpressionTree exprTree = PartFilterExprUtil.makeExpressionTree(expressionProxy, expr,
-                                                    getDefaultPartitionName(defaultPartitionName), conf);
+    final ExpressionTree exprTree = expr.length != 0 ? PartFilterExprUtil.makeExpressionTree(
```
Comment (marton-bod) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:3942]: Can you explain why this change was needed?
